Adds long range backpack radio

Novacat opened the pull request:
:cl: Novacat
   rscadd: Adds a long range backpack radio. Currently admin spawn only.
   /:cl:

MistakeNot4892 commented on code/game/objects/items/devices/radio/radio.dm:
```diff
@@ -473,6 +474,13 @@
 	//THIS IS TEMPORARY. YEAH RIGHT
 	if(!connection)	return 0	//~Carn
+
+	if(bluespace_radio)
```
````
	var/target_levels = bluespace_radio ? GetConnectedZlevels(position.z) : list(0)
	return Broadcast_Message(connection, M, voicemask, pick(M.speak_emote),
					  src, message, displayname, jobname, real_name, M.voice_name,
					  filter_type, signal.data["compression"], target_levels, connection.frequency, verb, speaking,
					  "[connection.frequency]", channel_color_presets["Menacing Maroon"])
````

MistakeNot4892 commented on code/game/objects/items/devices/radio/radio.dm:
```diff
@@ -511,7 +519,8 @@
 	if(!(0 in level))
 		var/turf/position = get_turf(src)
 		if(!position || !(position.z in level))
```
````
if((!position || !(position.z in level)) && !bluespace_radio)
    return -1
````

MistakeNot4892 commented on code/game/objects/items/devices/radio/radio.dm:
```diff
@@ -812,4 +821,132 @@
+		return
+
+	if(!slot_check())
+		to_chat(user, "<span class='warning'>You need to equip [src] before taking out [handset].</span>")
```
`SPAN_WARNING` here and below.

MistakeNot4892 commented on code/game/objects/items/devices/radio/radio.dm:
```diff
@@ -812,4 +821,132 @@
+/obj/item/device/subspaceradio/proc/slot_check()
+	var/mob/M = loc
+	if(!istype(M))
+		return 0 //not equipped
```
For clarity it might be better to `return FALSE` and `return TRUE` for 1.

MistakeNot4892 commented on code/game/objects/items/devices/radio/radio.dm:
```diff
@@ -812,4 +821,132 @@
+	if(ismob(handset.loc))
+		var/mob/M = handset.loc
+		if(M.drop_from_inventory(handset, src))
+			to_chat(user, "<span class='notice'>\The [handset] snap back into the main unit.</span>")
```
`snaps`. Also `SPAN_NOTICE`.

MistakeNot4892 commented on code/game/objects/items/devices/radio/radio.dm:
```diff
@@ -812,4 +821,132 @@
+	name = "subspace radio handset"
+	desc = "A large walkie talkie attached to the subspace radio by a retractable cord. It sits comfortably on a slot in the radio when not in use."
+	icon_state = "signaller"
+	slot_flags = null
```
`slot_flags = 0`

Novacat committed:
Adds long range backpack radio



:cl: Novacat
   rscadd: Adds a long range backpack radio. Currently admin spawn only.
   /:cl:
Recommended changes


Updates and fixes to_chat notices


Next time have the right repo open


A couple more overlooked span notices


Attempts to placate Travis


Incorrect indentations are a bitch

afterthought2 commented:
Marking this as last chance, as I don't see it being merged. We're pretty against cross-z radio communication generally. @Baystation12/developers , you have 3 days to speak up.

comma commented:
3 days habe passed rip
